src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java

```diff
@@ -41,4 +60,16 @@ public void addTo(final PageBuilder pageBuilder) {
             .optional(CaseData::getApplicant2HomeAddress, "applicant2SolicitorRepresented=\"No\"")
             .mandatory(CaseData::getApplicant2CorrespondenceAddress, "applicant2SolicitorRepresented=\"No\"");
     }
+
+    private AboutToStartOrSubmitResponse<CaseData, State> midEvent(
+        CaseDetails<CaseData, State> details,
+        CaseDetails<CaseData, State> detailsBefore
+    ) {
+        log.info("Mid-event callback triggered for Applicant2ServiceDetails");
+        return solicitorCreateApplicationService.setApplicant2SolOrganisationInfo(
```

Comment:
I saw this in another place, I'm not sure how I feel about things in the Service layer returning `AboutToStartOrSubmitResponse` objects. It kind of feels like that stuff should be at the callback handler level but it's not super important.

Reply:
will move it as a separate PR